packages/web/pages/api/stripe/webhook.ts

```diff
@@ -0,0 +1,107 @@
+import { MembershipSubscriptionPeriod } from '@journaly/j-db-client'
+import stripe from '../../../nexus/utils/stripe'
+import { getClient } from '../../../nexus/utils'
+
+const handler = async (req: any, res: any) => {
+  const db = getClient()
+  const event = req.body
+
+  const updateStripeSubscription = async (subscriptionId: string) => {
+    const stripeSubscription = await stripe.subscriptions.retrieve(subscriptionId)
+    let expiresAt = stripeSubscription.current_period_end * 1000
+
+    // Apply a grace period of 2 days to 'active' subscriptions
+    if (stripeSubscription.status === 'active') {
+      expiresAt += 24 * 60 * 60 * 1000 * 2
+    }
+    
+    await db.membershipSubscription.update({
+      where: {
+        stripeSubscriptionId: subscriptionId,
+      },
+      data: {
+        expiresAt: new Date(expiresAt)
```

Comment:
let's also update `stripeSubscription` (the JSON field) while we're here.

Reply:
✅ Done!